From dbce1905d674ec627ab6876770dda6f952580d7b Mon Sep 17 00:00:00 2001
From: Dave Kessler <activethrasher00@gmail.com>
Date: Sat, 2 Jan 2016 15:51:05 -0500
Subject: [PATCH 1/6] Reduce sync safety levels from 3 to 1

Benchmarks	Without		With
Seconds		21.39		12.13

* This patch improves the sqlite speed by ~44%
Thanks to dorimanx for helping me out with this patch.

Change-Id: I8547ade072cb4010ca3268131ee36d09860aaa04
Signed-off-by: mydongistiny <jaysonedson@gmail.com>
---
 dist/sqlite3.c | 33 ++++++++++-----------------------
 1 file changed, 10 insertions(+), 23 deletions(-)

diff --git a/dist/sqlite3.c b/dist/sqlite3.c
index 3322b52..8808fbe 100644
--- a/dist/sqlite3.c
+++ b/dist/sqlite3.c
@@ -51575,15 +51575,9 @@ SQLITE_PRIVATE void sqlite3PagerSetFlags(
   unsigned pgFlags      /* Various flags */
 ){
   unsigned level = pgFlags & PAGER_SYNCHRONOUS_MASK;
-  if( pPager->tempFile ){
-    pPager->noSync = 1;
-    pPager->fullSync = 0;
-    pPager->extraSync = 0;
-  }else{
-    pPager->noSync =  level==PAGER_SYNCHRONOUS_OFF ?1:0;
-    pPager->fullSync = level>=PAGER_SYNCHRONOUS_FULL ?1:0;
-    pPager->extraSync = level==PAGER_SYNCHRONOUS_EXTRA ?1:0;
-  }
+  pPager->noSync = 1;
+  pPager->fullSync = 0;
+  pPager->extraSync = 0;
   if( pPager->noSync ){
     pPager->syncFlags = 0;
   }else if( pgFlags & PAGER_FULLFSYNC ){
@@ -52897,18 +52891,11 @@ act_like_temp_file:
   pPager->memDb = (u8)memDb;
   pPager->readOnly = (u8)readOnly;
   assert( useJournal || pPager->tempFile );
-  pPager->noSync = pPager->tempFile;
-  if( pPager->noSync ){
-    assert( pPager->fullSync==0 );
-    assert( pPager->extraSync==0 );
-    assert( pPager->syncFlags==0 );
-    assert( pPager->walSyncFlags==0 );
-  }else{
-    pPager->fullSync = 1;
-    pPager->extraSync = 0;
-    pPager->syncFlags = SQLITE_SYNC_NORMAL;
-    pPager->walSyncFlags = SQLITE_SYNC_NORMAL | (SQLITE_SYNC_NORMAL<<2);
-  }
+  pPager->noSync = 1;
+  assert( pPager->fullSync==0 );
+  assert( pPager->extraSync==0 );
+  assert( pPager->syncFlags==0 );
+  assert( pPager->walSyncFlags==0 );
   /* pPager->pFirst = 0; */
   /* pPager->pFirstSynced = 0; */
   /* pPager->pLast = 0; */
@@ -100928,7 +100915,7 @@ static void attachFunc(
 #endif
     sqlite3BtreeLeave(pNew->pBt);
   }
-  pNew->safety_level = SQLITE_DEFAULT_SYNCHRONOUS+1;
+  pNew->safety_level = 1;
   pNew->zDbSName = sqlite3DbStrDup(db, zName);
   if( rc==SQLITE_OK && pNew->zDbSName==0 ){
     rc = SQLITE_NOMEM_BKPT;
@@ -146360,7 +146347,7 @@ static int openDatabase(
   ** database it is OFF. This matches the pager layer defaults.  
   */
   db->aDb[0].zDbSName = "main";
-  db->aDb[0].safety_level = SQLITE_DEFAULT_SYNCHRONOUS+1;
+  db->aDb[0].safety_level = 1;
   db->aDb[1].zDbSName = "temp";
   db->aDb[1].safety_level = PAGER_SYNCHRONOUS_OFF;
 
-- 
2.17.1

